lib/Mojolicious/Guides/Testing.pod

```diff
@@ -0,0 +1,754 @@
+  Host: myapp.example.com
+  Authorization: Basic bWU6c2VjcmV0bHkK=
+
+  {"level":"50%"}
+
+Or:
+
+  POST /blender/spin-blades
+  Host: myapp.example.com
+  Authorization: Basic bWU6c2VjcmV0bHkK=
+
+  {"speed":"whip"}
+
+Or:
+
+  POST /thermostat/heat
+  Host: myapp.example.com
+  Authorization: Basic bWU6c2VjcmV0bHkK=
+
+  {"temperature":"23°C"}
+
+To test kind of web application, this we could do something like this for
+every test:
+
+  $t->post_ok('http://myapp.example.com/thermostat/heat',
+              {Authorization => 'Basic bWU6c2VjcmV0bHkK='},
+              json => {temperature => "23°F"})
+    ->status_is(200);
+
+But what we'd like to do is to make the tests as simple as regular
+L<Mojolicious> tests:
+
+  $t->post_ok('/thermostat/heat',
+              json => {temperature => "23°F"});
+
+To do this we're going to override L<Test::Mojo>'s constructor and L<post_ok>
+methods. The constructur will accept a C<base_url>, C<username>, and
+C<password> parameters. The C<post_ok> method will add the C<Authorization>
+header for us.
+
+Because L<Test::Mojo>'s constructor has some shortcuts built into it for
+passing configuration data to Mojolicious applications, we can't invoke
+C<SUPER::new> without removing our custom arguments first:
+
+  package Test::MyApp;
+  use Mojo::Base 'Test::Mojo';
+  use Mojo::Util 'b64_encode';
+
+  has [my @attr = qw/base_url username password/];
+
+  sub new {
+    my $class = shift;
+    my %args  = @_;
+    my %app   = ();
+
+    # Remove custom attributes
+    @app{@attr} = delete @args{@attr};
+
+    my $self = $class->SUPER::new(%args);
+
+    # Set custom attributes
+    $self
+      ->base_url($app{base_url})
+      ->username($app{username})
+      ->password($app{password});
+
+    return $self;
+  }
+
+  sub post_ok {
+    my $self    = shift;
+    my $url     = $self->base_url . shift;
+    my $headers = (ref $_[0] eq 'HASH' ? shift : {});
+    $headers->{Authorization} //= "Basic " .
+      b64_encode join(':' => $self->username, $self->password), '';
+
+    $self->SUPER::post_ok($url, $headers, @_);
+  }
+
+  1;
+
+Our version of C<post_ok> works just like L<Test::Mojo>'s, but it adds the
+C<Authorization> header our web application needs for every call. To use our
+new L<Test::Mojo> subclass:
+
+  use Test::MyApp;
+  use Test::More;
+
+  my $t = Test::MyApp->new(base_url => 'http://myapp.example.com',
+                           username => 'me',
+                           password => 'secretly');
+
+  $t->post_ok('/blender/faster',
+              json => {speed => 'frappé'})
+    ->status_is(200);
+
+  # Use invalid password
+  $t->password('sorry!');
+  $t->post_ok('/blender/faster',
+              json => {speed => 'frappé'})
+    ->status_is(401);
+
+  done_testing();
+
+Great! Our little L<Test::Mojo>-based class allows us to make authorized
+requests to our deployed web application, and the test assertion (C<post_ok>)
+preserves compatibility with L<Test::Mojo>.
+
+=head1 MORE
```

Comment:
I have given some MVC training using Mojolicious in the past that might be a good start for that sort of tutorial. I agree that it's out of scope for this, but would be valuable information nonetheless.